meilisearch/tests/index/test_index.py

```diff
@@ -73,60 +93,71 @@ def test_get_or_create_index(self):
         assert len(documents) == 1
         index_2.delete()
         with pytest.raises(Exception):
-            self.client.get_index(index_3).info()
+            self.client.get_index(index_3)
 
     def test_get_or_create_index_with_primary_key(self):
         """Test get_or_create_index method with primary key"""
         index_1 = self.client.get_or_create_index('books', {'primaryKey': self.index_uid4})
         index_2 = self.client.get_or_create_index('books', {'primaryKey': 'some_wrong_key'})
+        assert index_1.primary_key == self.index_uid4
         assert index_1.get_primary_key() == self.index_uid4
+        assert index_2.primary_key is None
```

Comment:
I don't understand why this works. If line 100 works normally, the "books" index should have a `primaryKey` set to `self.index_uid4`
If afterwards we do a call to `get_or_create_index('books', {'primaryKey': 'some_wrong_key'})` (line 101), the index books exists already... shouldn't the primary key still be set to `self.index_uid4` in the response?

Reply:
This is a limitation use-case.
Here, when we do a `get_or_create_index` for `index2`, we really do a `get` (not a `create`) and the `get` is not really a "get" (I mean with an HTTP call), but a simple instance of `Index` (with no HTTP call), so we don't know the primary key value.
If we really do a get (with HTTP call) in `get_or_create_index` this method would do two HTTP calls in case of index creation. But we can change this despite this two-calls limitation.

Here would be the new version of `get_or_create_index`:

```py
def get_or_create_index(self, uid, options=None):
        try:
            index_instance = self.get_index(uid)
        except MeiliSearchApiError as err:
            if err.error_code != 'index_not_found':
                raise err
            index_instance = self.create_index(uid, options)
        return index_instance
```

Worst case: two calls during index creation
"Classic" case: one call during the `get`

We have to choose between a sometimes-two-calls `get_or_create_index` with an always-accurate `primary_key` attribute or a always-one-call `get_or_create_index` with a not-always-accurate `primary_key` attribute. WDYT @bidoubiwa @eskombro?

Reply:
Honestly, I'm a little undecided. I really don't like the fact that you could have a wrong `primary_key` stored in the Index instance after actually calling this method who is kind of "supposed" to retrieve your index. And `get_or_create_index()` is not a method called 1000 times per hour probably, so indeed it is interesting to do fewer calls, but I can't decide if it is worth doing it here. I let you decide since I don't really have a structured solid opinion on it :)

Reply:
The double HTTP call would be only done if the index is not created: it means if you do not delete your indexes but you launch the script twice (that may be why you chose to use `get_or_create_index` instead of `create_index`), you don't have the double call at the second launch. So, I would hope the double call is not a huge inconvenience (I hope 🤞).

Reply:
I prefer doing an HTTP call if the word `get` is used. Because `get` refers to an HTTP method.
I do not remember what we decided in meeting but I prefer: 
>  a sometimes-two-calls get_or_create_index with an always-accurate primary_key attribute

Reply:
> It means you will not have the double call at the second launch, but only at the first.

Yes, but it means in your example there would be 1001 calls instead of... 1

Reply:
But now that we have `index('...')` method, this one could be used to avoid the HTTP calls if that's what you want.

> I prefer doing an HTTP call if the word `get` is used. Because `get` refers to an HTTP method.
> I do not remember what we decided in meeting but I prefer:
> 
> > a sometimes-two-calls get_or_create_index with an always-accurate primary_key attribute

I am mostly inclined to think the same

Reply:
> Yes, @curquiza, but it means in your example there would be 1001 calls instead of... 1

You're right haha! 😅 

Let's try with a "sometimes-two-calls get_or_create_index with an always-accurate primary_key attribute" then, if some users seem it's not as they expect, we will reconsider it. I'm going to update the function.